### DISCUSSION TITLE
Couldn't render Animal Logic’s ALab

### DISCUSSION THREAD

**Discussion by freed-kenneth** [Autodesk/maya-usd#2084] [General] [answered]
HI,

I'm using Maya 2022.3 with 0.15.0 USD plugin. Currently I'm building an USD pipeline and heavily referencing the ALab scene' complexities.
I can load the ALab scene and see it with Viewport 2.0 , but when I switch the renderer to GL or Arnold I can see nothing.
The scene is totally fine in USDView and Houdini.

Does anyone know what is the issue?

**Comment by suekieza** [comment]
Yes we are having a lot of issues as well

**Comment by quentinBirrer** [comment]
We were testing within Maya 2022.2 with MayaUSD-0.14.0 and we had the same issue.

Can you confirm you have a generic message 
`// Error: Failed verification: ' result == MStatus::kSuccess ' //`
when loading it too?

We found out that issues get capture in this error, and usually we couldn't tell exactly what was the problem and where.
We had multiple issues like (non exhaustive):
- Do not understand relative path in shaders.
- Ignore meterPerUnits metadata.
- Do not understand None typed prim (E.g.  Def "myPrim").

It's hard to know where, in a complex asset like the ALab scene (or our own), Maya fails to understand a piece of data, and therefore look like it's not loading a functional USD for the Hydra delegates.

**Comment by santosg87** [comment, marked as answer]
this seems to be only happening with the "preview" for the geo_vis variant compositions. on my side i have been using the "default" compositions and hydra storm handles it ok on Maya.

this seems to be something with how the rprims are set. if you open any specific asset and try to select the geometry, Maya throws this error:
```
// Error: _InitRepr() should be called for repr refined on prim /HdMayaViewportRenderer/_HdMaya_HdStormRendererPlugin_000002156898BED0/_Delegate_HdMayaSceneDelegate_0_000002156898BED0/rprims/chemistry_bottle01/chemistry_bottle01Shape/ProxyDelegate_chemistry_bottle01Shape_0000021542E31F30/root/GEO/chemistry_bottle_hrc/bottle_1_geo. // 
// Error: Rprim has no draw items for repr refined // 
```

after digging a bit, it seems to be caused by the purpose set on the mesh level. simply resetting the purpose in maya gets things updated.

there are a couple workarounds for the time being (none of which is ideal, dare i say)
1. set your geo_vis variants sets to "default" 
2. manually add a "default" purpose to the meshes, instead of letting the prim set it - this seems to force a redraw and displays things.

I am curious on what makes usdview and Maya behave differently here. I will log a bug on our side to investigate this. 

one thing to note here is that hydra storm in maya is  an experimental plugin. :)

**Comment by santosg87** [comment]
@quentinBirrer the error you mentioned regarding "failed verification" is something else, we identified it and it should be mostly fixed on the latest builds of the plugin. :)

**Reply by quentinBirrer** [reply]
Good to know, we are transitioning now to Maya-2022.3 and MayaUSD-0.16.0.
The purpose error is also present in 2022.2 .

**Comment by stkrake** [comment]
The problem seems to be that MayaToHydra does not set the "Display Purposes" (`HdxTaskController::SetRenderTags()`) to
"Proxy" or "Render". If you uncheck "Proxy" and "Render" in usdview's "Display Purposes"-menu the ALab disappears, too.

If I see it right, MtohRenderOverride never calls `HdxTaskController::SetRenderTags()` at all and leaves the default of "geometry".
VP2 works because it uses a different code path than MtoH at this point.

Technically one could maybe implement a simple usdview-like "Display Purposes" selector for MtoH. Although I don't know whether 
this would collide with more advanced use cases (per-render item settings and such).

**Comment by stkrake** [comment]
Just saw that there already exists issue #1381 which may cover this.
